In [1]:
print("Hello World")

Hello World


In [ ]:
#Many Values to Multiple Variables
# cars = ["Swift", "Verna", "Seltos", "Legender"]
# a,b,c,d=cars
# print(a)
# print(b)
# print(c)
# print(d)

#variables
# x = "Python"
# y = "is"
# z = "awesome"
# print(x, y, z)

#variable type
# x = 5
# print(type(x))

#printing table of 5
# for i in range(1,11,1):  # the range will run till 10 only
#   print(5*i)

# counting odd and even numbers between 1 to 30
# count_even=0
# count_odd=0
# for i in range(1,31,1):
#   if(i%2==0):
#     count_even+=1
#   else:
#     count_odd+=1

# print("Count of even numbers are: ", count_even)
# print("Count of odd numbers are: ", count_odd)

#user input
# value=input("Enter value of your choice: ")
# print(value)

#function calling
# def add(x,y):
#   result=x+y
#   return result
# add(2,3)

# factorial
# n=int(input("Enter number: "))
# fact=1
# for i in range(1,n+1,1):
#   fact=fact*i

# print(fact)

#factorial using recursion
# n=int(input())
# def factorial(n):
#   if n<=1:
#     return 1
#   else:
#     return n*factorial(n-1)

# print(factorial(n))

#print fibonacci using user input with recursion
# num=int(input("Enter number: "))
# num=5
# def fib(num):
#   if(num<=1):
#     return num
#   else:
#     return (fib(num-1)+fib(num-2))

# for i in range(num):
#   print(fib(i),end=" ")


#print fibonacci using user input without recursion
# num=int(input("Enter number: "))
# a=0
# b=1
# print(a)
# print(b)
# for i in range(1,num-1,1):
#   c=a+b
#   print(c)
#   a=b
#   b=c


In [2]:
n = 5
for i in range(1,5):
    for j in range(1,i+1):
        print(j, end="")
    n += 1
        
    print("")

1
12
123
1234


In [3]:
#pattern printing
n = int(input())
spaces = n-1
count = 1
for i in range(1, n+1, 1):
    for space in range(spaces-i+1, 0, -1):
        print(" ", end = "")

    for num in range(1, i+1, 1):
        print(count, end = " " )
        count += 1
        
    print()
    

4
   1 
  2 3 
 4 5 6 
7 8 9 10 


# Assignment

# 1.
Given an array 'arr' containing 'n' elements, rotate this array left once and return it. Rotating the array left by one means shifting all elements by one place to the left and moving the first element to the last position in the array.

Example:

Input: 'a' = 5, 'arr' = [1, 2, 3, 4, 5] Output: [2, 3, 4, 5, 1]

Explanation: We moved the 2nd element to the 1st position, and 3rd element to the 2nd position, and 4th element to the 3rd position, and the 5th element to the 4th position, and move the 1st element to the 5th position.

In [6]:
a=5
list1=[1,2,3,4,5]
temp=list1[0]
for i in range(a-1):
    list1[i]=list1[i+1]
    
list1[a-1]=temp

for i in range(a):
    print(list1[i],end=", ")

2, 3, 4, 5, 1, 

# 2.
You are given an array 'arr' of length 'n', consisting of integers. A subarray is a contiguous segment of an array. In other words, a subarray can be formed by removing 0 or more integers from the beginning and 0 or more integers from the end of an array. Find the sum of the subarray (including empty subarray) having maximum sum among all subarrays. The sum of an empty subarray is 0.

Example :

Input: 'arr' = [1, 2, 7, -4, 3, 2, -10, 9, 1]

Output: 11

Explanation: The subarray yielding the maximum sum is [1, 2, 7, -4, 3, 2].

In [9]:
arr=[1, 2, 7, -4, 3, 2, -10, 9, 1]
curr_sum=0
max_sum=0
for i in range(len(arr)):
    curr_sum+=arr[i]
    max_sum=max(curr_sum,max_sum)
    if(curr_sum<0):
        curr_sum=0

print(max_sum)

11


# 3.
Write a function to reverse a string without using any built-in reverse functions.

In [2]:
str1="KUSHAGRA"
str2=""
for i in range(len(str1)-1,-1,-1):
    str2+=str1[i]
    
print(str2)

# string = "abcd"
# reverse_string = string[::-1]
# reverse_string

ARGAHSUK


# 4
Write a function to check if two strings are anagrams of each other.

Example:

Input 1: "listen"

Input 2: "silent"

Output: Are Anagrams: True

Explanation

function first removes spaces and converts both input strings to lowercase to make the comparison case-insensitive. Then, it checks if the sorted list of characters in both strings is the same. If they are the same, the strings are anagrams.

The function returns True for this example because the characters in "listen" can be rearranged to form "silent." If the characters were different or had different frequencies, the function would return False.

In [17]:
def check_anagram(string1, string2):
    return sorted(string1) == sorted(string2)     

    
string1 = "listen"
string2 = "silent"
    
check_anagram(string1, string2)

True

# Practice Questions... 

# 1.
Write a program to print all the unique characters in a String. For instance, if the input string is “abcb”, the output will be the characters ‘a’ and ‘c’ as they are unique. The character ‘b’ repeats twice and so it will not be printed.

Input: abcab

Output: c

In [25]:
str1 = input()
my_dict = {}

for i in str1:
    if i in my_dict:
        my_dict[i] += 1
    else:
        my_dict[i] = 1

for key, value in my_dict.items():
    if value == 1:
        print(key)


mkmkml
l


# 2.
Given three values - Start Fahrenheit Value (S), End Fahrenheit value (E) and Step Size (W), you need to convert all Fahrenheit values from Start to End at the gap of W, into their corresponding Celsius values and print the table.

Sample Input : 120 200 40

Sample Output : 120 48 160 71 200 93

In [37]:
def fahrenheit_to_celsius(start_f, end_f, step):
    result = []
    while start_f <= end_f:
        celsius = int((start_f - 32) * 5/9)
        result.append(celsius)
        start_f = start_f + step      
    return result 

start_f = int(input())
end_f = int(input())
step = int(input())

fahrenheit_to_celsius(start_f, end_f, step)

5
95
5


[-15, -12, -9, -6, -3, -1, 1, 4, 7, 10, 12, 15, 18, 21, 23, 26, 29, 32, 35]

# 3.
write a program to input an integer 'n' and print the sum of all its even digits and the sum of all its odd digits separately.

Digits mean numbers, not places! That is, if the given integer is "132456", even digits are 2, 4, and 6, and odd digits are 1, 3, and 5.

Example :

Input: 'n' = 132456 Output: 12 9

Explanation:

The sum of even digits = 2 + 4 + 6 = 12 The sum of odd digits = 1 + 3 + 5 = 9

In [38]:
n = int(input())
sum_even = 0
sum_odd = 0

while n > 0:
    digit = n % 10
    if digit % 2 == 0:
        sum_even = sum_even + digit
    else:
        sum_odd = sum_odd + digit
    n = n//10
    
print("Sum of even Numbers",sum_even)
print("Sum of odd Numbers",sum_odd)

123456
Sum of even Numbers 12
Sum of odd Numbers 9


# 4.
Write a program that takes a number as input and prints all its factors except 1 and the number itself.. If the number has only two factors (1 and the number itself), then the program should print -1.

Sample Input: 8

Sample Output: 2 4

Explanation of Sample Output 1 : The factors for the number excluding 1 and itself are 2 and 4, so the output is 2 4.

In [42]:
num = int(input())
c=0
for i in range(2, num, 1):
    if num % i == 0:
        print(i,end=" ")
        c+=1


if c == 0:
    print(-1)


11
-1


# 5.
Write a program to find x to the power n (i.e. x^n). Take x and n from the user. You need to print the answer.

Note : For this question, you can assume that 0 raised to the power of 0 is 1

Sample Input : 3 4

Sample Output : 81

In [43]:
x = int(input("Enter the Number: "))
n = int(input("Enter the power: ")) 
pow(x,n)

Enter the Number: 2
Enter the power: 4


16

# 6.
Write a program to generate the reverse of a given number N. Print the corresponding reverse number.

Note : If a number has trailing zeros, then its reverse will not include them. For e.g., reverse of 10400 will be 401 instead of 00401.

Sample Input 1 : 1234

Sample Output 1 : 4321

In [1]:
num = int(input("Enter the number: "))

rev = 0
while num != 0:
    digit = num % 10
    rev = rev * 10 + digit
    num = num // 10
    
print(rev)

Enter the number: 1200
21


# Pattern

# 1.
Print the following pattern for the given N number of rows.

Pattern for N = 4

A

BC

CDE

DEFG

In [53]:
n = int(input("Enter the number: "))
count = 0
for i in range(1,n+1):
    for j in range(1,i+1):
        print(chr(65 + count), end ="")
        count += 1     
    print("")

Enter the number: 4
A
BC
DEF
GHIJ


# 2.
Print the following pattern for the given N number of rows.

Pattern for N = 4

1

21

321

4321

In [54]:
n = int(input("Enter the number: "))

for i in range(1, n+1, 1):
    for j in range(i, 0 , -1):
        print(j, end="")
    print("")

Enter the number: 4
1
21
321
4321


# 3.
Print the following pattern for the given N number of rows.

Pattern for N = 3

A

BB

CCC

In [59]:
n = int(input("Enter the number: "))

for i in range(n+1):
    for j in range(i):
        print(chr(64 + i), end ="")    
    print("")

Enter the number: 4

A
BB
CCC
DDDD


# 5.
Print the following pattern for the given number of rows.

Pattern for N = 5

E

DE

CDE

BCDE

ABCDE

In [67]:
n=int(input())
for i in range(1,n+1,1):
    ch=69-i
    for j in range(1,i+1,1):
        print(chr(ch+j),end="")
    print()

4
E
DE
CDE
BCDE


In [ ]:
6.Write a program to print parallelogram pattern for the given N number of rows.

For N = 3
***
 ***
  ***

In [68]:
n = int(input())
spaces = n-1
count = 1
for i in range(1, n+1, 1):
    for space in range(1, i, 1):
        print(" ", end = "")

    for num in range(0, n, 1):
        print("*", end = "" )
        count += 1
        
    print()

4
****
 ****
  ****
   ****


# 7.

Pattern for N = 3
      1 
     12
    123

In [73]:
n = int(input())
spaces = n-1
count = 1
for i in range(1,n+1,1):
    for space in range(1, n-i+1, 1):
        print(" ", end = "")

    for num in range(1, i+1, 1):
        print(num, end = "" )
        count += 1
        
    print()

9
        1
       12
      123
     1234
    12345
   123456
  1234567
 12345678
123456789


# 8.Write a program to print the pattern for the given N number of rows.

For N = 4

1357
3571
5713
7135

In [82]:
n=int(input())

for i in range(1,n+1,1):
    for j in range(i,n+1,1):
        print(2*j-1,end="")
        
    for k in range(1,i,1):
        print(2*k-1,end="")
        
    print()

4
1357
3571
5713
7135
